lib/miq_expression.rb

```diff
@@ -1606,8 +1616,10 @@ def self.ruby_for_date_compare(col_ruby, col_type, tz, op1, val1, op2 = nil, val
 
   def to_arel(exp, tz)
     operator = exp.keys.first
-
-    field = Field.parse(exp[operator]["field"]) if exp[operator].kind_of?(Hash) && exp[operator]["field"]
+    if exp[operator].kind_of?(Hash)
+      field = Field.parse(exp[operator]["field"]) if exp[operator]["field"]
+      exp[operator]["value"] = Field.parse(exp[operator]["value"]).arel_attribute if exp[operator]["value"] && Field.valid_field?(exp[operator]["value"])
```

Comment:
@alongoldboim I am :+1: for incremental change, but this concerns the public API for MiqExpression and will be hard to change once committed. I think it's important to get this part right the first time